### ISSUE TITLE
Support for cloudflare kv?

### ISSUE THREAD

**Issue by ImLunaHey** [jcs224/hono_sessions#20]
any chance cloudflare kv could also be added?

**Comment by ImLunaHey** [comment]
i believe this is the correct implementation. the only thing missing is adding the cloudflare worker types as a dep.

```ts
import { SessionData, Store } from 'hono-sessions';

export class CloudflareoKvStore implements Store {
  kv: KVNamespace;
  collectionName: string;

  constructor(kv: KVNamespace, collectionName = 'sessions') {
    this.kv = kv;
    this.collectionName = collectionName;
  }

  async getSessionById(sessionId: string) {
    return await this.kv.get<SessionData>(`${this.collectionName}/${sessionId}`);
  }

  async createSession(sessionId: string, initialData: SessionData) {
    await this.kv.put(`${this.collectionName}/${sessionId}`, JSON.stringify(initialData));
  }

  async deleteSession(sessionId: string) {
    await this.kv.delete(`${this.collectionName}/${sessionId}`);
  }

  async persistSessionData(sessionId: string, sessionData: SessionData) {
    await this.kv.put(`${this.collectionName}/${sessionId}`, JSON.stringify(sessionData));
  }
}
```

**Comment by jcs224** [comment]
I likely won't add KV as a supported option, since it's an eventually-consistent storage, which doesn't really jive with the heavy read-write nature of sessions, as per the KV docs:

https://developers.cloudflare.com/kv/concepts/how-kv-works/

![image](https://github.com/user-attachments/assets/0561641a-8ee0-42ea-b4c3-c100b65a0a4a)

There is a Cloudflare D1 driver in the codebase, although un-documented currently. I've kept it under the radar since when I wrote it, D1 was still in beta. That looks to no longer be the case so maybe I'll revisit it one of these days.
